var/spack/repos/builtin/packages/py-multiqc/package.py

```diff
@@ -31,5 +31,5 @@ class PyMultiqc(PythonPackage):
     depends_on('py-pyyaml', type=('build', 'run'))
     depends_on('py-simplejson', type=('build', 'run'))
     depends_on('py-requests', type=('build', 'run'), when='@1.5:')
-    depends_on('py-enum34', type=('build', 'run'), when='@1.5:')
+    depends_on('py-enum34', type=('build', 'run'), when='@1.5: ^python@:3.3')
```

Comment:
```suggestion
    depends_on('py-enum34', type=('build', 'run'), when='@1.4:1.5 ^python@:3.3')
    depends_on('py-enum34', type=('build', 'run'), when='@1.3')
```

Reply:
Well, I was going to ask how the heck you knew that, when it wasn't mentioned in the one `setup.py` file that I checked, but I took a deep breath and started unpacking the various tarballs and checking their `setup.py` files and sure enough 1.4 and 1.5 have:

```python
if sys.version_info < (3, 4):
    install_requires.append('enum34')
```

while the v1.3 `setup.py` has it unconditionally listed in it's `install_requires`.

Is there something magical w/in `enum34` that makes is safe to use (a no-op, or ...) with `python@3.4:` or something about multiqc@1.3 that ensures `python@:3.3, or ????

Do you have magically tooling for dumping setup.py info into Spack package format?

Reply:
> Is there something magical w/in enum34 that makes is safe to use (a no-op, or ...) with python@3.4: or something about multiqc@1.3 that ensures `python@:3.3, or ????

Not that I know of. The problem here is that for v1.3, if you don't provide enum34, setuptools will download and install it itself, which won't work on an air-gapped network. I imagine it's a no-op if you look at the multiqc source code.

Personally, I would remove the `conflicts()` from `py-enum34`. It isn't needed, and may not work for Python 3.4+, but it also won't harm anything if you install it and never import it.

> Do you have magical tooling for dumping setup.py info into Spack package format?

Nope, just brute force and determination 😄

Reply:
> Nope, just brute force and determination 😄

Blunt hammers of the Internet, **unite**!